zephyr/esp32c6/src/wifi/esp_wifi_adapter.c

```diff
@@ -0,0 +1,1076 @@
+	._timer_disarm = timer_disarm_wrapper,
+	._timer_done = timer_done_wrapper,
+	._timer_setfn = timer_setfn_wrapper,
+	._timer_arm_us = timer_arm_us_wrapper,
+	._wifi_reset_mac = wifi_reset_mac_wrapper,
+	._wifi_clock_enable = wifi_clock_enable_wrapper,
+	._wifi_clock_disable = wifi_clock_disable_wrapper,
+	._wifi_rtc_enable_iso = esp_empty_wrapper,
+	._wifi_rtc_disable_iso = esp_empty_wrapper,
+	._esp_timer_get_time = esp_timer_get_time,
+	._nvs_set_i8 = nvs_set_i8,
+	._nvs_get_i8 = nvs_get_i8,
+	._nvs_set_u8 = nvs_set_u8,
+	._nvs_get_u8 = nvs_get_u8,
+	._nvs_set_u16 = nvs_set_u16,
+	._nvs_get_u16 = nvs_get_u16,
+	._nvs_open = nvs_open_wrapper,
+	._nvs_close = nvs_close,
+	._nvs_commit = nvs_commit,
+	._nvs_set_blob = nvs_set_blob,
+	._nvs_get_blob = nvs_get_blob,
+	._nvs_erase_key = nvs_erase_key,
+	._get_random = os_get_random,
+	._get_time = get_time_wrapper,
+	._random = os_random,
+	._slowclk_cal_get = esp_coex_common_clk_slowclk_cal_get_wrapper,
+	._log_write = esp_log_write_wrapper,
+	._log_writev = esp_log_writev_wrapper,
+	._log_timestamp = k_uptime_get_32,
+	._malloc_internal =  malloc_internal_wrapper,
+	._realloc_internal = realloc_internal_wrapper,
+	._calloc_internal = calloc_internal_wrapper,
+	._zalloc_internal = zalloc_internal_wrapper,
+	._wifi_malloc = wifi_malloc,
+	._wifi_realloc = wifi_realloc,
+	._wifi_calloc = wifi_calloc,
+	._wifi_zalloc = wifi_zalloc_wrapper,
+	._wifi_create_queue = wifi_create_queue_wrapper,
+	._wifi_delete_queue = wifi_delete_queue_wrapper,
+	._coex_init = coex_init_wrapper,
+	._coex_deinit = coex_deinit_wrapper,
+	._coex_enable = coex_enable_wrapper,
+	._coex_disable = coex_disable_wrapper,
+	._coex_status_get = coex_status_get_wrapper,
+	._coex_wifi_request = coex_wifi_request_wrapper,
+	._coex_wifi_release = coex_wifi_release_wrapper,
+	._coex_wifi_channel_set = coex_wifi_channel_set_wrapper,
+	._coex_event_duration_get = coex_event_duration_get_wrapper,
+	._coex_pti_get = coex_pti_get_wrapper,
+	._coex_schm_status_bit_clear = coex_schm_status_bit_clear_wrapper,
+	._coex_schm_status_bit_set = coex_schm_status_bit_set_wrapper,
+	._coex_schm_interval_set = coex_schm_interval_set_wrapper,
+	._coex_schm_interval_get = coex_schm_interval_get_wrapper,
+	._coex_schm_curr_period_get = coex_schm_curr_period_get_wrapper,
+	._coex_schm_curr_phase_get = coex_schm_curr_phase_get_wrapper,
+	._coex_register_start_cb = coex_register_start_cb_wrapper,
+#if SOC_PM_MODEM_RETENTION_BY_REGDMA
+	._regdma_link_set_write_wait_content = regdma_link_set_write_wait_content_wrapper,
+	._sleep_retention_find_link_by_id = sleep_retention_find_link_by_id_wrapper,
+#endif
+	._coex_schm_process_restart = coex_schm_process_restart_wrapper,
+	._coex_schm_register_cb = coex_schm_register_cb_wrapper,
+    ._coex_schm_flexible_period_set = coex_schm_flexible_period_set_wrapper,
+    ._coex_schm_flexible_period_get = coex_schm_flexible_period_get_wrapper,
+	._magic = ESP_WIFI_OS_ADAPTER_MAGIC,
+};
+
+static void esp_wifi_free(void *mem)
```

Comment:
Whats the reason for burying this call down below? Could we move it to other heap calls?

Reply:
Sure, moved up to the same position as done in other adapters.